src/Build.UnitTests/BackEnd/SdkResolverLoadingStrategy_Tests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.UnitTests;
+using Microsoft.Build.UnitTests.BackEnd;
+using Xunit;
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    public class SdkResolverLoadingStrategy_Tests
+    {
+        [Fact]
+        public void AssertDefaultLoadingStretegyReturnsDefaultResolver()
+        {
+            var loader = new SdkResolverLoadingStrategy();
+            var log = new StringBuilder();
+            var logger = new MockLoggingService(message => log.AppendLine(message));
+            var bec = new BuildEventContext(0, 0, 0, 0, 0);
+
+            var resolvers = loader.LoadResolvers(logger, bec, new MockElementLocation("file"));
+
+            Assert.Equal(1, resolvers.Count);
+            Assert.Equal(typeof(DefaultSdkResolver), resolvers[0].GetType());
+        }
+
+        [Fact]
+        public void VerifySdkResolverFileDiscoveryPattern()
+        {
+            var root = FileUtilities.GetTemporaryDirectory();
+            try
+            {
+                // Valid pattern is root\(Name)\(Name).dll. No other files should be considered.
+                var d1 = Directory.CreateDirectory(Path.Combine(root, "Resolver1"));
+                var d2 = Directory.CreateDirectory(Path.Combine(root, "NoResolver"));
+
+                // Valid.
+                var f1 = Path.Combine(d1.FullName, "Resolver1.dll");
+
+                // Invalid, won't be considered.
+                var f2 = Path.Combine(d1.FullName, "Dependency.dll");
+                var f3 = Path.Combine(d2.FullName, "InvalidName.dll");
+                var f4 = Path.Combine(d2.FullName, "NoResolver.txt");
+
+                File.WriteAllText(f1, string.Empty);
+                File.WriteAllText(f2, string.Empty);
+                File.WriteAllText(f3, string.Empty);
+                File.WriteAllText(f4, string.Empty);
+
+                var strategy = new SdkResolverLoadingStrategy();
+                var files = strategy.FindPotentialSdkResolvers(root);
```

Comment:
Ah, never mind. This is an internal implementation detail and is appropriately named.